iib/workers/tasks/git_utils.py

```diff
@@ -288,3 +293,202 @@ def revert_last_commit(
+    :rtype: Dict[str, str]
+    :raises IIBError: If GitLab API call fails.
+    """
+    # Extract GitLab information from repository URL
+    api_url, project_path = _extract_gitlab_info(repo_url)
+
+    # GitLab API endpoint for creating merge requests
+    api_url = f"{api_url}/projects/{quote_plus(project_path)}/merge_requests"
+
+    headers = {'Content-Type': 'application/json', 'Authorization': f'Bearer {git_token}'}
+
+    payload = {
+        'source_branch': source_branch,
+        'target_branch': target_branch,
+        'title': f'IIB: Update for request id {request_id}',
+        'description': f'Automated merge request created by IIB for request {request_id}',
+        'remove_source_branch': True,
+        'squash': True,
+    }
+
+    try:
+        log.info("Creating merge request via GitLab API for project %s", project_path)
+        response = requests_session.post(api_url, headers=headers, json=payload, timeout=30)
+
+        if not response.ok:
+            log.error(
+                'Failed to create merge request. Status: %d, Response: %s',
+                response.status_code,
+                response.text,
+            )
+            raise IIBError(f'Failed to create merge request: {response.status_code}')
+
+        mr_data = response.json()
+        mr_id = str(mr_data['iid'])
+        mr_url = mr_data['web_url']
+
+        log.info("Successfully created merge request %s: %s", mr_id, mr_url)
+
+        return {'mr_id': mr_id, 'mr_url': mr_url, 'source_branch': source_branch}
+
+    except requests.RequestException as e:
+        log.exception("Error creating merge request via GitLab API")
+        raise IIBError(f'GitLab API request failed: {str(e)}')
+
+
+@instrument_tracing(span_name="workers.tasks.git_utils.close_mr")
+def close_mr(mr_details: Dict[str, str], repo_url: str) -> None:
+    """
+    Close a merge request on GitLab repository.
+
+    :param dict mr_details: Dictionary containing MR details (mr_id, mr_url, source_branch).
+    :param str repo_url: Git repository URL.
+    :raises IIBError: If GitLab API call fails.
+    """
+    mr_id = mr_details.get('mr_id')
+    if not mr_id:
+        raise IIBError("Missing mr_id in mr_details")
+
+    # Get GitLab token for API access
+    _, git_token = get_git_token(repo_url)
+
+    # Close merge request via GitLab API
+    _close_gitlab_mr(repo_url, git_token, mr_id)
+
+
+def _close_gitlab_mr(repo_url: str, git_token: str, mr_id: str) -> None:
+    """
+    Close a merge request using GitLab API.
+
+    :param str repo_url: Git repository URL.
+    :param str git_token: GitLab access token.
+    :param str mr_id: Merge request ID.
+    :raises IIBError: If GitLab API call fails.
+    """
+    # Extract GitLab information from repository URL
+    api_url, project_path = _extract_gitlab_info(repo_url)
+
+    # GitLab API endpoint for updating merge requests
+    api_url = f"{api_url}/projects/{quote_plus(project_path)}/merge_requests/{mr_id}"
```

Comment:
Does the project reflect the index image (each index image, one project)?

Reply:
yes, it would essentially be `namespace/repo` extracted from the git url.